gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-http-proxy/src/main/java/io/gravitee/plugin/endpoint/http/proxy/HttpProxyEndpointConnector.java

```diff
@@ -114,13 +130,18 @@ private boolean isGrpc(final HttpRequest httpRequest) {
         }
     }
 
-    @Override
-    protected void doStop() {
-        if (httpClientFactory != null) {
-            httpClientFactory.close();
-        }
-        if (grpcHttpClientFactory != null) {
-            grpcHttpClientFactory.close();
+    private Completable handleException(Throwable throwable, HttpExecutionContext ctx) {
+        if (
```

Comment:
You should also consider that if it is already an `InterruptionFailureException`, you should let it go